### DISCUSSION TITLE
How to use the schema registry with the kafka preset?

### DISCUSSION THREAD

**Discussion by ahjashish** [orlangure/gnomock#664] [Q&A]
Hi guys, when I start a kafka container with gnomock, I see there are other services including Schema Registry that comes up but I am unable to figure out how to communicate with it. Is there any examples on doing that for integration tests that anyone already worked on?
I want to have a schema for my kafka event and check if the event pertains to the given schema but I am not clear on the api to use to create schemas and query them with the service that comes along with kafka preset?
The workflow I want to test looks like this:
1. Have Kafka with a topic
2. Have a protobuf schema for the event to be published on the topic
3. Produce an event with that schema and consume the event reading the schema
4. Produce an event with compatible schema with backward compatibility and consume compatible event with the schema

**Comment by orlangure** [comment]
Hi @ahjashish,

I never needed to interact with the schema registry, but from looking at [the image repo](https://github.com/lensesio/fast-data-dev) I see that it has another exposed port - `:8081` for it. At this moment Kafka preset does not publish this port, but it should be very easy to add. Another entry should be added to the named ports (schema registry with port 8081). The code is here: https://github.com/orlangure/gnomock/blob/master/preset/kafka/preset.go#L77-L88

Would you like to try that and submit a PR?😼

**Comment by orlangure** [comment]
Anyway, now Kafka registry is available when requested using preset options:
https://github.com/orlangure/gnomock/releases/tag/v0.22.0
